src/SlidingWindow.cpp

```diff
@@ -292,7 +394,31 @@ class SlidingWindowOnFunctionAndLoop : public IRMutator {
                     stmt = LetStmt::make(n, max(var, b[dim_idx].max), stmt);
                 }
             }
+
+            // Guard producers against running on expanded bounds.
+            Expr orig_loop_min = Variable::make(Int(32), loop_var + ".loop_min.orig");
+            Expr bounded_loop_var = max(orig_loop_min, loop_var_expr);
+            Expr bounded_min = substitute(loop_var, bounded_loop_var, min_required);
+            stmt = guard_producer(stmt, func, dim_idx, bounded_min, Expr());
+
             return stmt;
+        } else if (!find_produce(op, func.name())) {
+            // The producer might have expanded the loop before the min to warm
+            // up the window. This consumer doesn't contain a producer that might
+            // be part of the warmup, so guard it with an if to only run it on
+            // the original loop bounds.
+            Expr loop_var_expr = Variable::make(Int(32), loop_var);
+            Expr orig_loop_min_expr = Variable::make(Int(32), loop_var + ".loop_min.orig");
+            Expr guard = likely_if_innermost(loop_var_expr >= orig_loop_min_expr);
```

Comment:
Prefer <= to >= for the same reason as above, unless you think it makes it confusing to read to the degree that it's worth a few extra nanoseconds of compile time.